nomad/nextcloud.nomad

```diff
@@ -0,0 +1,261 @@
+        labels {
+          liquid_task = "snoop-ncsync-rabbitmq"
+        }
+      }
+      resources {
+        network {
+          port "amqp" {}
+        }
+      }
+      service {
+        name = "snoop-ncsync-rabbitmq"
+        port = "amqp"
+      }
+    }
+
+    task "tika" {
+      driver = "docker"
+      config {
+        image = "logicalspark/docker-tikaserver"
+        port_map {
+          http = 9998
+        }
+        labels {
+          liquid_task = "snoop-ncsync-tika"
+        }
+      }
+      resources {
+        network {
+          port "tika" {}
+        }
+      }
+      service {
+        name = "snoop-ncsync-tika"
+        port = "tika"
+      }
+    }
+
+    task "pg" {
+      driver = "docker"
+      config {
+        image = "postgres:9.6"
+        volumes = [
+          "/opt/hoover/volumes/snoop-pg--ncsync/data:/var/lib/postgresql/data",
+        ]
+        labels {
+          liquid_task = "snoop-ncsync-pg"
+        }
+        port_map {
+          pg = 5432
+        }
+      }
+      env {
+        POSTGRES_USER = "snoop"
+        POSTGRES_DATABASE = "snoop"
+      }
+      resources {
+        network {
+          port "pg" {}
+        }
+      }
+      service {
+        name = "snoop-ncsync-pg"
+        port = "pg"
+      }
+    }
+  }
+ 
+    group "workers" {
+    count = 2
+
+    task "snoop" {
+      driver = "docker"
+      config {
+        image = "liquidinvestigations/hoover-snoop2:liquid-nomad"
+        args = ["./manage.py", "runworkers -s"]
+        volumes = [
+          "/opt/hoover/gnupg:/opt/hoover/gnupg",
+          "/opt/hoover/collections/ncsync:/opt/hoover/snoop/collection",
+          "/opt/hoover/snoop-blobs/ncsync:/opt/hoover/snoop/blobs",
+        ]
+        labels {
+          liquid_task = "snoop-ncsync-worker"
+        }
+      }
+      env {
+        SNOOP_COLLECTION_ROOT = "collection"
+        SNOOP_TASK_PREFIX = "ncsync"
+        SNOOP_ES_INDEX = "ncsync"
+      }
+      template {
+        data = <<EOF
+            {{- if keyExists "liquid_debug" }}
+              DEBUG = {{ key "liquid_debug" }}
+            {{- end }}
+            SNOOP_DB = postgresql://snoop:snoop@
+              {{- range service "snoop-ncsync-pg" -}}
+                {{ .Address }}:{{ .Port }}
+              {{- end -}}
+              /snoop
+            SNOOP_ES_URL = http://
+              {{- range service "hoover-es" -}}
+                {{ .Address }}:{{ .Port }}
+              {{- end }}
+            SNOOP_TIKA_URL = http://
+              {{- range service "snoop-ncsync-tika" -}}
+                {{ .Address }}:{{ .Port }}
+              {{- end }}
+            SNOOP_AMQP_URL = amqp://
+              {{- range service "snoop-ncsync-rabbitmq" -}}
+                {{ .Address }}:{{ .Port }}
+              {{- end }}
+          EOF
+        destination = "local/snoop.env"
+        env = true
+      }
+      resources {
+        memory = 512
+      }
+    }
+  }
+  group "api" {
+    task "snoop" {
```

Comment:
Do you really need a `snoop` task in the nextcould job? I think the snoop should be separate, and configured to pull data from nextcloud via nomad *services*.

Reply:
I did it like this so that it is the whole package in one file.  I also set them up with the same script. We could change that but I saw them as intertwined so much that bringing it all up includes snoop and nextcloud.

Reply:
Yes, but we already have a nomad config for snoop, it would be problematic to maintain two in parallel. I'd rather have the one be configurable so it can pull data from nextcloud.

Reply:
You're right about that.  
Come to think of it, I'll add an issue to the board about collection creation with the nomad setup.